osf/models/preprint_service.py

```diff
@@ -139,7 +161,7 @@ def set_subjects(self, preprint_subjects, auth):
         self.save(old_subjects=old_subjects)
 
     def set_primary_file(self, preprint_file, auth, save=False):
-        if not self.node.has_permission(auth.user, ADMIN):
+        if not self.has_permission(auth.user, 'ADMIN'):
```

Comment:
I think the tests are failing for this method (and others) in part because you haven't yet set contributor's permissions on the preprint, they just still exist on the node